lib/rules/boolean-prop-naming.js

```diff
@@ -226,7 +227,9 @@ module.exports = {
       if (!node || !Array.isArray(args)) {
         return;
       }
-      args.filter((arg) => arg.type === 'ObjectExpression').forEach((object) => validatePropNaming(node, object.properties));
+      args.forEach((object) => object.type === 'ObjectExpression'
+        && validatePropNaming(node, object.properties)
+      );
```

Comment:
how meaningful is this specific change? the engine should be able to optimize this down to the equivalent for loop.

Reply:
Good question.
I'm not convinced v8 can optimise a filter and forEach into a for loop - some (admittedly slapdash and micro) benchmarking with node v14.15.3 iterating many times over an array shows a singular forEach with a built-in conditional as reliably faster than a filter, forEach chain:
```
$ ITERATIONS=100 SIZE=10000 node arraytest-filter-for-each.js
filter, forEach: time taken 58.374 ms
$ ITERATIONS=100 SIZE=10000 node arraytest-just-for-each.js
just forEach: time taken 43.283 ms
$ ITERATIONS=100 SIZE=10000 node arraytest-for-of-loop.js
for of loop: time taken 41.101 ms
```
The singular forEach is consistently faster - depending on the arrangement of the code seems to be around 20 -> 40% faster.
Going with more iterations but a smaller array shows similar results too, just in case:
```
$ ITERATIONS=10000 SIZE=100 node arraytest-filter-for-each.js
filter, forEach: time taken 45.790 ms
$ ITERATIONS=10000 SIZE=100 node arraytest-just-for-each.js
just forEach: time taken 40.437 ms
$ ITERATIONS=10000 SIZE=100 node arraytest-for-of-loop.js
for of loop: time taken 36.737 ms
```
Here's the gist:
https://gist.github.com/willheslam/d45a1d183d5d778d82c5113ffb8a4ab8

Whether this specific change is meaningful, I don't honestly have concrete answer - it depends on the frequency the function is called, the average size of `args` (and how many of them fulfil the condition)
and also how much the surrounding environment is generating memory pressure.

Benchmarking each individual change is difficult, given the noise and iteration time and how some effects of memory allocation are only obvious in concert, so as a blanket rule tried to find any quick, localised opportunities to reduce memory allocation, where the change didn't massively impact the flow of the code.